docs/3_ObjectRecognition.md

```diff
@@ -0,0 +1,486 @@
+
+Use "feh" to visualize the photo : 
+
+```bash
+sudo apt install feh
+```
+
+<!-- TODO #### Run yolov8 on the rasp -->
+
+##### Predict an image (not a live video)
+
+First, update packages list, install pip and upgrade to latest.
+
+```bash
+sudo apt update
+sudo apt install python3-pip -y
+pip install -U pip
+```
+
+Then install ultralytics (they developed yolov8).
+
+```bash
+pip install ultralytics[export]
+```
+
+Reboot the device
+
+```bash
+sudo reboot
+```
+
+Convert Model to NCNN and Run Inference
+
+```bash
+# Export a YOLOv8n PyTorch model to NCNN format
+yolo export model=yolov8n.pt format=ncnn  # creates 'yolov8n_ncnn_model'
+
+# Run inference with the exported model
+yolo predict model='yolov8n_ncnn_model' source='https://ultralytics.com/images/bus.jpg'
+```
+
+The output of that is the source image with bounding box that identifies each object. 
+
+> NB : you can take whatever "source" you want, even locally. 
+
+⚠️ Make sure the image sizes are 640x640 maximum.
+
+![](assets/9f32524e6765c337a39003b06.png)
+
+> Pretty good, huh 😎
+
+##### Inference with camera
+
+We need to initiate a TCP stream with `rpicam-vid` from the connected camera so that we can use this stream URL as an input when we are inferencing later. Execute the following command to start the TCP stream.
+
+```bash
+rpicam-vid -n -t 0 --inline --listen -o tcp://127.0.0.1:8888
+```
+
+In other tab (so open another ssh connection) : 
+
+```bash
+yolo predict model=yolov8n.pt source="tcp://127.0.0.1:8888"
+```
+
+With that, we didn't see what the camera recoded but we had a text feedback like : 2 persons 0.8 probability, 1 smartphone 0.2 probability. And it was quite accurate. 
+⚠️ **Issue :** But the performance wasn't good. We had 1/4 FPS (1 image every 4 seconds). Yolov8 did not work for us. 
+❓ What model to choose now ??? Time for benchmark
+
+##### Looking for another model
+
+We did some more research and we needed something lighter than yolov8. 
+
+We found yolov5 and mobilenet and tested them following that doc : https://github.com/raspberrypi/picamera2
+
+Mobilenet was the more effective so we pre-
+
+**2. Fine-tune yolov8 model on our computers**
+
+So we've picked Mobilenetv3_Large model, and now what ? 
+Now, let's fine-tune it. 
+
+DON'T follow [Tensorflow's documentation](https://www.tensorflow.org/tutorials/images/transfer_learning?hl=fr). Follow the [Keras'](https://keras.io/guides/transfer_learning/) one instead. 
+
+#### Import your own dataset
+
+Importing your own dataset can be tricky, we tested many things : 
+
+- **Google drive** : A dataset is too big to be imported on a drive, it may fill it. 
+- **kaggle**: You need to be authentified.
+- **Hugging face** : 
+- **Google Cloud Storage** : 
+
+<!-- **3. Run our fine-tuned model on the rasp** -->
```

Comment:
We still struggle to fine-tune the mode. We seem to have compatibility problems between our fine-tuned model in tflite and the model expected by the object detection script. We don't know where the problem comes from